### ISSUE TITLE
Support for XLM-RoBERTa 

### ISSUE THREAD

**Issue by hieudx149** [oriram/spider#5]
if I use XLM-RoBERTa instead of BERT, all I need is to change model_name_or_path to "xlm-roberta-base" or Do I need to do something else?

**Comment by oriram** [comment]
Hey @r1ckC139 ,
If you do that, please make sure that the model and tokenizer are loaded properly.
I'd love to hear whether it went successfully.

**Comment by hieudx149** [comment]
Hi @oriram,
Thanks for your quick response, I pre-trained spider with my own dataset (in Vietnamese), so all Wikipedia Vietnamese page is just about 1.5 million pages, and after chunking all page to 100word-passages I have 1.5 GB of pre-trained data, I only have one  A100 80G GPU (in your paper you guys use eight GPU). Based on your experience with spiders, can you suggest to me some important hyperparameters like batch size, learning rate and training steps with the hardware I have? I'd like to tell you if my work is successful.

**Comment by oriram** [comment]
I'd use the largest batch size I can for as many training steps as I can.
Regarding batch size, I guess it depends on the model so perhaps play around and see how the training loss behaves :)

**Comment by hieudx149** [comment]
Hi @oriram 
if I use roberta, it must be fairseq_roberta, it seems like DPR doesn't support hf_roberta, do you have any way to use hf_roberta model? Another question, In your paper, you said you use only one encoder for both query and context, but DPR uses two separate encoders, so in pre-trained you use only one, and in fine-tune stage you use two encoders like DPR, am I right?

**Comment by oriram** [comment]
1. I think it should be pretty simple to implement a `hf_roberta` model.
2. You are right, we are using shared weights during pretraining and untie the weights during fine-tuning. However, there is little to no difference between the two modes at fine-tuning. The only reason we did that is to be consistent with DPR's setting.

**Comment by hieudx149** [comment]
Hi @oriram,
I'd like to tell you about my work (pre-trained spider with hf xlm-roberta for Vietnamese), it works but the performance is slightly lower than bm25, I guess this result is because of the lower pre-trained dataset than English, and I need to tune the hyperparameters. Anyway thanks for your help, maybe I can release my code for pre-trained spider for another language that's not English.

**Comment by oriram** [comment]
Hi @r1ckC139 ,
Sorry for missing your message.
Did you pretrain for long enough?

**Comment by hieudx149** [comment]
@oriram i set update_steps up to 200000, i used 2 A100 80G and i took about 2 days

**Comment by oriram** [comment]
I see.. 
I think in that case, you might want to check if during the training process the model even improved.
If so, it might be the case that the batch size is not large enough.
